pkg/gadgets/trace/exec/tracer/tracer_test.go

```diff
@@ -0,0 +1,219 @@
+//go:build linux
+// +build linux
+
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracer
```

Comment:
It's recommended to perform black-box tests when possible, so testing only exported identifiers. To ensure this, you can add `_test` to the test package name and then you have to import `tracer` package as you would usually do. This also serves great for the documentation purposes as code examples.